docs/api/ReactWrapper/render.md

```diff
@@ -29,5 +29,5 @@ function Bar() {
 
 ```jsx
 const wrapper = mount(<Bar />);
-expect(wrapper.find(Foo).render().find('.in-foo')).to.have.lengthOf(1);
```

Comment:
i'm not sure about this one - it seems useful to show that you can call `.render()` on any sub-wrapper.

Reply:
`.render() `can call on any sub-wrapper, but `.render() `seems doesn't return the Bar component's highest `<div> `as well as `Foo `component's. I used `HTML() `on `wrapper.render()`and `wrapper.find(Foo).render()`, it didnt' render the first `<div>` and `<Foo> `component didn't render anything HTML output, so the test failed and didn't find the `.in-foo`. I put a wrapping `<div>` around the `.in-foo`on my end, it did rendered the `.in-foo `classname. It seems calling `.render()` may need to have a wrapping tag around the HTML.

Reply:
hmm - I would expect `wrapper.find(Foo).render()` to be the same as `render(<Foo />)`. Is it not?

Reply:
I tested the above example with the following test, the `wrapper.find(Foo).render()` was same as `render(<Foo />)` and `wrapper.find(Foo).render()` had one length, but it didn't find the `.in-foo` node.
```
it('is wrapper.find(Foo).render() == render(<Foo />) ',()=>{
                const wrapper = mount(<Bar />);
                const renderWrapper=render(<Foo/>)
                console.log("a",wrapper.find(Foo).render().html())  // not outputing HTML 
                expect(wrapper.find(Foo).render()).toEqual(renderWrapper)   //test pass
                expect(wrapper.find(Foo).render()).toHaveLength(1)   // test pass
                 expect(wrapper.find(Foo).render().find(".in-foo")).toHaveLength(1)  // test failed
                
            })
```

Reply:
These examples are actually what's used in the tests: https://github.com/enzymejs/enzyme/blob/master/packages/enzyme-test-suite/test/shared/methods/render.jsx#L13-L42

Based on your last test, this is actually correct - if it *is* `in-foo`, `.find` will only search the children, and thus won't find any.

Reply:
If `.find` will only search the children in this case, then should we make .in-foo in the `.render `example as children like the below example?
```	
function Foo( ) {
  return (
     <div>
         <div className="in-foo" />
     </div>
);
}
```

Reply:
`.filter` or `.is` could work

Reply:
Based on your suggestion, I modified the test to this. Is it better?
`expect(wrapper.find(Foo).render().filter(".in-foo")).to.Have.Length(1);`

Reply:
Sure, that works! Would it still work if the `.filter` came before the `.render()`?

Reply:
when the `.filter `came before the `.render()`,  it didn't search the children and didn't find any nodes, so the test failed. I used `.children()` to allow the `.filter` to search the matched node in the children. It passed. Does this look good? What do you think? 
`expect(wrapper.find(Foo).filter(".in-foo").render()).to.Have.Length(1);  //test failed. Error: HTML means to run on 1 node but 0 node found`
` expect(wrapper.find(Foo).children().filter(".in-foo").render()).to.Have.Length(1);  // test passed`